### PR TITLE
docs: Replace type `mixed` in system.

### PR DESCRIPTION
**Pull request by ping-yee**
**Description**
See #6310 

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/BaseModel.php [hunk 1]
```diff
@@ -1711,7 +1711,7 @@ protected function transformDataToArray($data, string $type): array
      *
      * @param string $name Name
      *
-     * @return mixed
+     * @return array|bool|int|object|string|null
```
Comment (kenjis) [inline system/BaseModel.php:1714]:
```suggestion
     * @return array|bool|float|int|object|string|null
```

File: system/Model.php [hunk 2]
```diff
@@ -792,7 +792,7 @@ protected function objectToRawArray($data, bool $onlyChanged = true, bool $recur
      *
      * @param string $name Name
      *
-     * @return mixed
+     * @return array|BaseBuilder|bool|int|object|string|null
```
Comment (kenjis) [inline system/Model.php:795]:
```suggestion
     * @return array|BaseBuilder|bool|float|int|object|string|null
```

File: system/Model.php [hunk 3]
```diff
@@ -821,7 +821,7 @@ public function __isset(string $name): bool
      * Provides direct access to method in the builder (if available)
      * and the database connection.
      *
-     * @return mixed
+     * @return array|BaseBuilder|bool|float|int|object|string|null
```
Comment (kenjis) [inline system/Model.php:824]:
```suggestion
     * @return $this|array|BaseBuilder|bool|float|int|object|string|null
```

File: system/Model.php (hunk 2 above)
Comment (ping-yee) [reply system/Model.php]: Done.

File: system/Model.php (hunk 3 above)
Comment (ping-yee) [reply system/Model.php]: Done.

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
@ping-yee Thank you!

**Pull request merged**